### ISSUE TITLE
Logically equivalent plans produced different results geo_mean

### ISSUE THREAD

**Issue by laithsakka** [facebookincubator/velox#7202]
### Description

https://github.com/facebookincubator/velox/actions/runs/6608207379/job/17946601811

### Error Reproduction

NA

### Relevant logs

```bash
NA
```

**Comment by mbasmanova** [comment]
@xumingming Would you help look into this?

**Comment by xumingming** [comment]
@mbasmanova Yeah, I am digging.

The failed plan is  the variant: `partial + final + abandon-partial-aggregation`:

```

I1023 03:57:34.174831 20565 AggregationFuzzer.cpp:269] Testing plan #1 with forced abandon-partial-aggregation
I1023 03:57:34.174842 20565 AggregationFuzzer.cpp:814] Executing query plan: 
-- Aggregation[FINAL [g0, g1] a0 := geometric_mean("a0")] -> g0:SMALLINT, g1:SMALLINT, a0:DOUBLE
  -- Aggregation[PARTIAL [g0, g1] a0 := geometric_mean(ROW["c0"]) mask: m0] -> g0:SMALLINT, g1:SMALLINT, a0:ROW<"":DOUBLE,"":BIGINT>
    -- Values[1000 rows in 10 vectors] -> c0:DOUBLE, m0:BOOLEAN, g0:SMALLINT, g1:SMALLINT
I1023 03:57:34.183866 24057 Task.cpp:983] All drivers (1) finished for task test_cursor 1511 after running for 8 ms.
I1023 03:57:34.183879 24057 Task.cpp:1665] Terminating task test_cursor 1511 with state Finished after running for 8 ms.
I1023 03:57:34.184033 20565 AggregationFuzzer.cpp:844] [ROW ROW<g0:SMALLINT,g1:SMALLINT,a0:DOUBLE>: 746 elements, no nulls]
../../velox/exec/tests/utils/QueryAssertions.cpp:1029: Failure
Failed
Expected 746, got 746
12 extra rows, 12 missing rows
10 of extra rows:
	null | 3914 | 1
	null | 10885 | 0.3350434339635034
	178 | 1817 | 0.5166842341423035
	2029 | 8528 | 0.36296680057421327
	2029 | 11281 | 0.36296680057421327
	2029 | 20188 | 0.5166842341423035
	5613 | null | 0.22622404684197708
	5613 | 10885 | 0.3636051328296219
	6492 | 21869 | 0.12235425296239556
	12517 | 30434 | 0.5166842341423035

10 of missing rows:
	null | 3914 | null
	null | 10885 | null
	178 | 1817 | null
	2029 | 8528 | null
	2029 | 11281 | null
	2029 | 20188 | null
	5613 | null | null
	5613 | 10885 | null
	6492 | 21869 | null
	12517 | 30434 | null
```

**Comment by xumingming** [comment]
Update: it seems has something to do with `toIntermediate` fast path. If I remove `toIntermediate` function from `GeometricMeanAggregate` the issue will be gone(run fuzzer with the same seed that reproduce this issue).https://github.com/facebookincubator/velox/blob/2c0b7138473ff0e96b14c0e2707a7d7899086ec9/velox/functions/prestosql/aggregates/GeometricMeanAggregate.cpp#L38-L43

**Comment by kagamiori** [comment]
This is caused by the same bug as https://github.com/facebookincubator/velox/issues/7162.

**Comment by kagamiori** [comment]
Hi @xumingming, Thank you for taking the first step investigating the issue and help us narrow down the problem! FYI, it's caused by a bug in GroupingSet when resizing the result vector before calling toIntermediate(). A fix is made in #7224.

**Comment by xumingming** [comment]
@kagamiori Glad to help :) and thanks for the quick fixing!
